test/helpers/PlayerInteractionWrapper.js

```diff
@@ -309,6 +317,15 @@ class PlayerInteractionWrapper {
         });
     }
 
+    attachOpponentOwnedUpgrades(opponentOwnedUpgrades = []) {
+        opponentOwnedUpgrades.forEach((upgrade) => {
```

Comment:
Any chance we could use an explicit `for()` loop, `forEach` is a headache to debug